src/utils/proxy.js

```diff
@@ -0,0 +1,34 @@
+import { API_PUBLIC } from '../config';
+import axios from 'axios';
+const classPayloads = new Map();
```

Comment:
Perhaps can use VueX or other state management frameworks?

Reply:
I think we can keep it simple as it is a small example project.

Reply:
I think the issue is the result of `getClass` can be reused across the two pages, so it would be better to store them into either a single place. It can either be just a simple map or `vuex` store
Also `classPayloads` just store pending promises now so maybe we want to rewrite this to store data also

Reply:
Actually I have tried if I don't delete the finished promise it can store the result, too. But I want to keep it updates every time user press "load" to have refresh data ability.

Reply:
I think `refresh` should be an explicit that clear all map/store in this case, because we can expect much overlap between two pages

Reply:
Update it to reuse the result.